bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/esp32/WlanThermoEsp32Handler.java

```diff
@@ -0,0 +1,248 @@
+                    updateState(channel.getUID(), UnDefType.UNDEF);
+                }
+            } catch (InterruptedException e) {
+                logger.debug("Update interrupted. {}", e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            try {
+                State s = WlanThermoEsp32CommandHandler.getState(channelUID, data, settings);
+                updateState(channelUID, s);
+            } catch (WlanThermoException e) {
+                logger.debug("Could not handle command of type {} for channel {}!",
+                        command.getClass().toGenericString(), channelUID.getId());
+            }
+        } else {
+            if (WlanThermoEsp32CommandHandler.setState(channelUID, command, data)) {
+                logger.debug("Data updated, pushing changes");
+                scheduler.execute(this::push);
+            } else {
+                logger.debug("Could not handle command of type {} for channel {}!",
+                        command.getClass().toGenericString(), channelUID.getId());
+            }
+        }
+    }
+
+    private void push() {
+        if (data == null || this.thing.getStatus() != ThingStatus.ONLINE) {
+            return;
+        }
+
+        // Push update for sensor channels
+        for (org.openhab.binding.wlanthermo.internal.api.esp32.dto.data.Channel c : data.getChannel()) {
+            try {
+                String json = gson.toJson(c);
+                URI uri = config.getUri("/setchannels");
+                int status = httpClient.POST(uri).content(new StringContentProvider(json), "application/json")
+                        .timeout(5, TimeUnit.SECONDS).send().getStatus();
+                if (status == 401) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "No or wrong login credentials provided. Please configure username/password for write access to WlanThermo!");
+                    break;
+                } else if (status != 200) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Failed to update channel "
+                            + c.getName() + " on device, Statuscode " + status + " on URI " + uri.toString());
+                    logger.debug("Payload sent: {}", json);
+                } else {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (TimeoutException | ExecutionException | URISyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Failed to update channel " + c.getName() + " on device: " + e.getMessage());
+            } catch (InterruptedException e) {
+                logger.debug("Push interrupted. {}", e.getMessage());
+            }
```

Comment:
You should return at this point to that you don't continue to query the other channels.
```suggestion
                logger.debug("Push interrupted. {}", e.getMessage());
                return;
            }
```